### ISSUE TITLE
add "flag menu" control during review

### ISSUE THREAD

**Issue by rickyribeiroo** [ankidroid/Anki-Android#15697]
**Is your feature request related to a problem? Please describe.**
During review, flags can be toggled individually by key, joystick / motion controller, or gesture. However, there's no control to just show the flags menu (similarly to "Add tag"). 

**Describe the solution you'd like**
I'd like to see a control option for "Add flag" that would show the flags menu, giving the user the option to then choose a flag from that menu during review.

**Comment by BrayanDSO** [comment]
I like it, but would show a dialog to pick a flag instead. Either a vertical one or a 3x3 grid one, with the flags icons and names

**Comment by divyesh-mali** [comment]
Hi @david-allison  
I'm excited to see this feature implemented in AnkiDroid. I'd be happy to contribute by working on the flag selection option during review. I've followed the discussion regarding menus vs dialogs, and I'm open to collaborating on the best approach. Can you please assign this issue to me? I'm eager to get started and discuss any specific details you may have.

**Comment by david-allison** [comment]
Not there (I presume in the reviewer). 

A flag belongs to a card, not to a note, so it shouldn't be a concern of the note editor

**Comment by rickyribeiroo** [comment]
Exactly, I mean openning the flag menu through a gesture / controller /
button when reviewing a card, the same way the option "add tag" exists in
the Gestures settings.

Thank you!

**Comment by divyesh-mali** [comment]
hey @rickyribeiroo 
can we connect on discord? I would like to discuss some things regarding this issue. My discord username is divyesh.mali

**Comment by david-allison** [comment]
Make sure you document the result of the conversation here, Discord isn't searchable

**Comment by divyesh-mali** [comment]
Summary of Discord Conversation with @rickyribeiroo :

I reached out to rickyribeiroo on Discord seeking clarification regarding the issue of adding flags during review in Ankidroid. rickyribeiroo explained the feature request, aiming to add functionality similar to the existing "Add tag" feature. He advised me to experiment with the "Add tag" feature to understand its functionality better. I agreed to proceed with implementing the "Add flag" feature and expressed gratitude for the guidance and I'll be trying my best to working on implementing this feature.

**Comment by david-allison** [comment]
(light mentoring): In future, just list what other people need to know

> It was explained that the functionality should be similar to the "Add tag" screen

**Comment by david-allison** [comment]
No worries! Education takes priority!

**Comment by RedLexa** [comment]
Hello! I am considering restarting work on this enhancement, and want to make sure I understand what is being requested here. 

> I like it, but would show a dialog instead of the options menu, since the menu wouldn't look natural if the toolbar is hidden. The dialog would be either a vertical one or a 3x3 grid one, with the flags icons and names, whichever looks better

I understood @BrayanDSO's suggestion of the dialog instead of the menu. But what exactly is the functionality in this dialog that isn't already present in the menu flag card  (that shows up next to the undo icon on the toolbar)? If I am understanding correctly two things are being asked: 

1.  The dialog would show the flags menu. 
2.  The user then choose a flag from that meny during review.

It seems to me that both of these functionalities are already implemented in this menu.
![image](https://github.com/ankidroid/Anki-Android/assets/109145734/97f3c69c-88c2-4c48-9eb5-003f34933f5e)

Apologies if I'm missing something obvious.

**Comment by david-allison** [comment]
The request is to make that menu mappable to a keyboard shortcut/gesture/controller in 'settings -> controls'

**Comment by RedLexa** [comment]
Thanks for the reply! I see now. My follow-up question would be: would the implementation of this dialog replace the current drop-down menu from the toolbar, even when it's not displayed as a consequence of the control being called?

**Comment by david-allison** [comment]
I doubt it's worth it, given the menu is simple and close to a user's finger/cursor when they make a press


Do a quick check using Fitts's law if you're included.

**Comment by RedLexa** [comment]
I'd like to work on this! @david-allison Can you please assign me?

**Comment by david-allison** [comment]
Thanks!!
